src/modules/roc_sndio/target_sox/roc_sndio/sox_sink.cpp

```diff
@@ -28,7 +29,7 @@ SoxSink::SoxSink(core::IAllocator& allocator, const Config& config)
         return;
     }
 
-    if (config.frame_size == 0) {
+    if (buffer_size_ == 0 && config.sample_rate != 0) {
```

Comment:
This alone won't work because I have to re-calculate the frame size with device rate from sox or pulseaudio. Will add that with another commit.

Reply:
Yeah, I guess we should delay `buffer_size_` calculation to `prepare()` call.

Reply:
So prepare() should be called in the constructor, yes?

Reply:
We pass frame_length in nanoseconds to constructor. We need to calculate frame_size in bytes from frame_length and sample_rate. However, we don't know sample_rate until open() is called.

open() calls prepare_() and open_(). open_() is the point from which we know sample_rate.

We can reorder prepare_() and open_() calls and calculate frame_size from frame_length in prepare_().

Something like that.